lib/ios/native-navigation/ReactNavigationImplementation.swift

```diff
@@ -51,16 +51,24 @@ class BlockBarButtonItem: UIBarButtonItem {
     self.target = self
   }
 
+  convenience init(barButtonSystemItem: UIBarButtonSystemItem) {
+    self.init(barButtonSystemItem: barButtonSystemItem, target: nil, action: #selector(barButtonItemPressed))
+    self.target = self
+  }
+
   convenience init(
     title: String?,
     image: UIImage?,
+    barButtonSystemItem: UIBarButtonSystemItem?,
     style: UIBarButtonItemStyle,
     enabled: Bool?,
     tintColor: UIColor?,
     titleTextAttributes: [String: Any]?
   ) {
     if let title = title {
       self.init(title: title, style: style)
+    } else if let barButtonSystemItem = barButtonSystemItem {
```

Comment:
Just a side note - in iOS, it's pretty much the fact that you can't have a mixed bar button type of a system vs one with a title/image. In the future, I'd love to support setting both a title AND an image which can be done through the `customView` initializer on `UIBarButtonItem`. That being said, it'd make sense to have this as the very first case, then if it isn't a system item, go on to title/image etc. Not a big deal now, yet. We can either bump this up now, or perhaps do it in the future, no big deal either way.